ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestContext.scala

```diff
@@ -3,30 +3,84 @@
 
 package com.daml.ledger.api.testtool.infrastructure
 
-import java.time.Instant
+import java.time.{Clock, Instant}
 
-import com.digitalasset.daml.lf.data.Ref
-import com.digitalasset.daml.lf.language.Ast
+import com.digitalasset.ledger.api.v1.active_contracts_service.{
+  GetActiveContractsRequest,
+  GetActiveContractsResponse
+}
+import com.digitalasset.ledger.api.v1.admin.party_management_service.AllocatePartyRequest
 import com.digitalasset.ledger.api.v1.command_service.SubmitAndWaitRequest
-import com.digitalasset.ledger.api.v1.commands.Command
-import com.digitalasset.ledger.api.v1.event.CreatedEvent
+import com.digitalasset.ledger.api.v1.commands.{Command, Commands}
+import com.digitalasset.ledger.api.v1.event.Event.Event.Created
+import com.digitalasset.ledger.api.v1.event.{CreatedEvent, Event}
 import com.digitalasset.ledger.api.v1.ledger_offset.LedgerOffset
+import com.digitalasset.ledger.api.v1.testing.time_service.{
+  GetTimeRequest,
+  GetTimeResponse,
+  SetTimeRequest
+}
 import com.digitalasset.ledger.api.v1.transaction.{Transaction, TransactionTree}
-import com.digitalasset.ledger.api.v1.value.Identifier
+import com.digitalasset.ledger.api.v1.transaction_filter.{
+  Filters,
+  InclusiveFilters,
+  TransactionFilter
+}
+import com.digitalasset.ledger.api.v1.transaction_service.{
+  GetLedgerEndRequest,
+  GetTransactionByIdRequest,
+  GetTransactionsRequest
+}
+import com.digitalasset.ledger.api.v1.value.{Identifier, Value}
 import com.digitalasset.ledger.client.binding.Primitive.Party
 import com.digitalasset.ledger.client.binding.{Contract, Primitive, Template, ValueDecoder}
+import com.digitalasset.platform.testing.{FiniteStreamObserver, SingleItemObserver}
+import com.google.protobuf.timestamp.Timestamp
+import io.grpc.stub.StreamObserver
+import scalaz.Tag
+import scalaz.syntax.tag._
 
 import scala.concurrent.duration.Duration
 import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+
+object LedgerTestContext {
+
+  private[this] def filter(templateIds: Seq[Identifier]): Filters =
+    new Filters(if (templateIds.isEmpty) None else Some(new InclusiveFilters(templateIds)))
+
+  private def transactionFilter(
+      parties: Seq[String],
+      templateIds: Seq[Identifier]): Some[TransactionFilter] =
+    Some(new TransactionFilter(Map(parties.map(_ -> filter(templateIds)): _*)))
 
-final class LedgerTestContext(
+  private def timestamp(i: Instant): Some[Timestamp] =
+    Some(new Timestamp(i.getEpochSecond, i.getNano))
+
+  private def timestampToInstant(t: Timestamp): Instant =
+    Instant.EPOCH.plusSeconds(t.seconds).plusNanos(t.nanos.toLong)
+
+  private def instantToTimestamp(t: Instant): Timestamp =
+    new Timestamp(t.getEpochSecond, t.getNano)
+
+  private val end = LedgerOffset(
+    LedgerOffset.Value.Boundary(LedgerOffset.LedgerBoundary.LEDGER_END))
+
+  private val defaultTtlSeconds = 30
+
+}
+
+private[infrastructure] final class LedgerTestContext(
+    val id: String,
```

Comment:
Is this the ledger ID?

Reply:
Yes, the only reason why I made this shorter it's because the binding of the context is usually called `ledger` and `ledger.id` seems a bit better than `ledger.ledgerId` but it's not that important, probably it's better to revert it.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/2573/commits/9fee1f3007d39c57c4533707639f2f158dac6028